hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/main/java/org/apache/hadoop/yarn/server/router/clientrm/FederationClientInterceptor.java

```diff
@@ -310,7 +314,9 @@ public GetNewApplicationResponse getNewApplication(
 
     routerMetrics.incrAppsFailedCreated();
     String errMsg = "Fail to create a new application.";
-    LOG.error(errMsg);
+    RouterAuditLogger.logFailure(user.toString(),
```

Comment:
If we will keep doing user.toString() we probably should add a method with user as an arg and do the toStirng there.

Reply:
On second thought, I think user.getShortUserName should be used instead of adding ugi info.